src/main/java/org/folio/rest/utils/PatronNoticeBuilder.java

```diff
@@ -0,0 +1,301 @@
+    }
+
+    final LoanPolicy loanPolicy = ctx.getLoanPolicy();
+
+    if (loanPolicy != null && loanPolicy.getRenewalsPolicy() != null) {
+      final RenewalsPolicy renewalsPolicy = loanPolicy.getRenewalsPolicy();
+
+      if (isTrue(renewalsPolicy.getUnlimited())) {
+        loanContext.put(NUMBER_OF_RENEWALS_ALLOWED, UNLIMITED);
+        loanContext.put(NUMBER_OF_RENEWALS_REMAINING, UNLIMITED);
+      } else {
+        Double numberAllowed = renewalsPolicy.getNumberAllowed();
+        int renewalLimit = numberAllowed == null ? 0 : numberAllowed.intValue();
+        loanContext.put(NUMBER_OF_RENEWALS_ALLOWED, Integer.toString(renewalLimit));
+
+        if (loan.getRenewalCount() != null) {
+          int renewalsRemaining = max(renewalLimit - loan.getRenewalCount(), 0);
+          loanContext.put(NUMBER_OF_RENEWALS_REMAINING, Integer.toString(renewalsRemaining));
+        }
+      }
+    }
+
+    return loanContext;
+  }
+
+  private static JsonObject buildFeeChargeContext(FeeFineNoticeContext ctx) {
+    final Account account = ctx.getAccount();
+    final JsonObject feeChargeContext = new JsonObject();
+
+    if (account == null) {
+      return feeChargeContext;
+    }
+
+    String paymentStatus = account.getPaymentStatus() == null ?
+      null : account.getPaymentStatus().getName();
+
+    feeChargeContext
+      .put("owner", account.getFeeFineOwner())
+      .put("type", account.getFeeFineType())
+      .put("paymentStatus", paymentStatus)
+      .put("amount", account.getAmount())
+      .put("remainingAmount", account.getRemaining());
+
+    final Metadata metadata = account.getMetadata();
+    if (metadata != null) {
+      String chargeDate = dateToString(metadata.getCreatedDate());
+
+      feeChargeContext
+        .put("chargeDate", chargeDate)
+        .put("chargeDateTime", chargeDate);
+    }
+
+    return feeChargeContext;
+  }
+
+  private static JsonObject buildFeeActionContext(FeeFineNoticeContext ctx) {
+    final Feefineaction action = ctx.getFeefineaction();
+    final JsonObject feeActionContext = new JsonObject();
+
+    if (action == null) {
+      return (feeActionContext);
+    }
+
+    String actionDate = dateToString(action.getDateAction());
+
+    feeActionContext
+      .put("type", action.getTypeAction())
+      .put("actionDate", actionDate)
+      .put("actionDateTime", actionDate)
+      .put("amount", action.getAmountAction())
+      .put("remainingAmount", action.getBalance())
+      .put("additionalInfo", getCommentsFromFeeFineAction(action));
+
+    return feeActionContext;
+  }
+
+  private static String getCommentsFromFeeFineAction(Feefineaction feefineaction){
+    String comments = Optional.ofNullable(feefineaction.getComments()).orElse(StringUtils.EMPTY);
+    return parseFeeFineComments(comments).getOrDefault(PATRON_COMMENTS_KEY, StringUtils.EMPTY);
+  }
+
+  static Map<String, String> parseFeeFineComments(String comments) {
```

Comment:
It seems like we already have the similar public static method here: org.folio.rest.utils.FeeFineActionCommentsParser#parseFeeFineComments